plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java

```diff
@@ -161,16 +165,20 @@ public Page getNextPage()
             close();
             return null;
         }
-        currentDataTable = dataTableList.pop();
 
+        long pageSizeBytes = 0L;
         PageBuilder pageBuilder = new PageBuilder(columnTypes);
-        // Note that declared positions in the Page should be the same with number of rows in each Block
-        pageBuilder.declarePositions(currentDataTable.getDataTable().getNumberOfRows());
-        for (int columnHandleIdx = 0; columnHandleIdx < columnHandles.size(); columnHandleIdx++) {
-            BlockBuilder blockBuilder = pageBuilder.getBlockBuilder(columnHandleIdx);
-            Type columnType = columnTypes.get(columnHandleIdx);
-            // Write a block for each column in the original order.
-            writeBlock(blockBuilder, columnType, columnHandleIdx);
+
+        while (dataTableList.size() > 0 && pageSizeBytes < maxSegmentPageSizeBytes) {
+            currentDataTable = dataTableList.pop();
+            pageSizeBytes += currentDataTable.getEstimatedSizeInBytes();
```

Comment:
this suggests that the maxSegmentPageSize is a soft-limit, correct? (if `currentDataTable bytes > maxSegmentPageSize`)

Reply:
Yes, if the page is larger then start a new page but allow the current page to be larger. I can rename to `segmentPageSizeBytesSoftLimit` wdyt?

Reply:
That or `targetSegmentPageSizeBytes`. Either would work.